locust_tasks/locustfile.py

```diff
@@ -166,49 +189,166 @@ def correct_address(self):
+        """
+        with self.client.post("/en/requests/access-code/enter-address/", {
+            'form-enter-address-postcode': self.case['postcode']
+        }, catch_response=True) as response:
+            self.address_to_select = extractAddress(response, self.case["uprn"])
+            verify_response('RequestUacSms-EnterAddress', self, response, 200, Page.SELECT_ADDRESS,
+                            self.case["postcode"])
 
-    @task(3)
+    @task
     def select_address(self):
-        # This code should arguable select one of the available addresses but running 
-        # with a fixed address doesn't seem to affect the success of the test
-        self.client.post("/request-access-code/select-address", {
-            'request-address-select': "{'uprn': '10023122452', 'address': hardcoded_address}"
-        })
+        """
+        POST uprn and whole address extracted as JSON from the HTML in previous task
+        """
+        #logger.info("Address: " + self.address_to_select)
+        with self.client.post("/en/requests/access-code/select-address/", {
+            'form-select-address': self.address_to_select
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-SelectAddress', self, response, 200, Page.ADDRESS_CORRECT, self.case["postcode"])
 
-    @task(4)
+    @task
     def confirm_address(self):
-        self.client.post("/request-access-code/confirm-address", {
-            'request-address-confirmation': 'yes'
-        })
+        """
+        POST 'yes' to confirm address
+        """
+        with self.client.post("/en/requests/access-code/confirm-address/", {
+            'form-confirm-address': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-ConfirmAddress', self, response, 200, Page.SELECT_METHOD, "Text message")
+
+    @task
+    def select_method(self):
+        """
+        POST 'sms' to select text message as method of sending UACs
+        """
+        with self.client.post("/en/requests/access-code/select-method/", {
+            'form-select-method': 'sms'
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-SelectMethod', self, response, 200, Page.ENTER_MOBILE)
 
-    @task(5)
+    @task
     def enter_mobile_number(self):
-        self.client.post("/request-access-code/enter-mobile", {
-            'request-mobile-number': '07714 330 933'
-        })
+        """
+        POST a phone number. Then use a section of the phone number (the last 3 digits) to verify the response.
+        """
+        self.phone_num = self.case["phone_number"]
+        #logger.info("Phone number: " + self.phone_num)
+        with self.client.post("/en/requests/access-code/enter-mobile/", {
+            'request-mobile-number': self.phone_num
+        }, catch_response=True) as response:
+            verify_response('RequestUacSms-EnterMobileNumber', self, response, 200, Page.CONFIRM_MOBILE, self.phone_num[-3:])
```

Comment:
Why just the last 3 digits?
We have the whole phone number so can't we verify it's all there?

Reply:
I tried it with the whole number at first but it gave me an error due to the spaces between the numbers in the particular number that I tried. I think it was rendered differently on the different HTML pages.

Reply:
I think it's worth investigating to see how it's rendered on the pages, and then building an expected string with the phone number that we've got.

Reply:
done and pushed